src/library/basetrackcache.cpp

```diff
@@ -145,16 +145,18 @@ void BaseTrackCache::setSearchColumns(const QStringList& columns) {
 }
 
 TrackPointer BaseTrackCache::lookupCachedTrack(TrackId trackId) const {
-    TrackCacheLocker cacheLocker(
-            TrackCache::instance().lookupById(trackId));
-    auto pTrack = cacheLocker.getTrack();
-    if (pTrack && pTrack->isDirty()) {
-        m_dirtyTracks.insert(trackId);
-        return pTrack;
-    } else {
-        m_dirtyTracks.remove(trackId);
-        return TrackPointer();
+    TrackPointer pTrack;
+    if (m_bIsCaching) {
```

Comment:
Is it correct that this is true for Mixxx tracks only, but not for third party libraries? 
Why? It looks like they have their own instance of BaseTrackCache. Is there his kind of legacy?
Can we either unify this to only one instance or inherit form BaseTracKCache as a base class?
I am not sure what makes actually sense here.